documentation/source/library_reference.rst

```diff
@@ -71,19 +71,18 @@ as the types used by cocotb's `simulator handles <#simulation-object-handles>`_.
 
 .. versionadded:: 2.0
 
-.. autoclass:: cocotb.types.Logic
+.. autoclass:: cocotb.types.logic.Logic
 
-.. autoclass:: cocotb.types.Bit
+.. autoclass:: cocotb.types.logic.Bit
 
-.. autoclass:: cocotb.types.Range
+.. autoclass:: cocotb.types.range.Range
     :members:
     :exclude-members: count, index
 
-.. autofunction:: cocotb.types.concat
-
-.. autoclass:: cocotb.types.Array
+.. autoclass:: cocotb.types.array.Array
```

Comment:
As I understand it, new versions of Sphinx will do the right thing with the old version given the contents of `types/__init__.py.`

Reply:
Doesn't seem to be happy locally, but I can always leave it the way it was and eventually it will start working?

Reply:
Not a big deal either way - if you can get it working how it was, then Sphinx will show the type name as `cocotb.types.Array` instead of `cocotb.types.array.Array`.

Reply:
I returned it to the way it was. The worst thing is that some type annotations aren't linking correctly, for types that are not even a page above/below.

Reply:
It's already released(4.x), but our rtd setup isn't finding it

Reply:
We are calling out `sphinx ~= 3.1` in `documentation/requirements.txt`. It switches to Sphinx 4.1.1 if I remove that, but it errors out. I'll chase that down.